### ISSUE TITLE
SkipLibCheck fails to work in multi-project set-up

### ISSUE THREAD

**Issue by NoelAbrahams** [microsoft/TypeScript#36503]
**TypeScript Version:**  3.8. Beta

**Search Terms:**
skiplibcheck

**Steps to reproduce**
1. Extract the attached zip file to a folder and open `repro.sln` in Visual Studio (I'm using 2019)
2. Expand the `npm` folder on project B and install the missing `@azure/storage-blob` package.
3. Rebuild the solution.

**Expected behavior:**
No build errors.

**Actual behavior:**
2 errors:
> error TS2740: Build:Type 'Buffer' is missing the following properties...
> error TS2345: Build:Argument of type 'ErrnoException' is not assignable...

**Explanation**
We want have a custom definition for NodeJS. The npm package imported in Project B also has its own (incomplete) definition. So, we've added `skiplibcheck` to all the projects.

Project B builds fine by itself.

Project C references some NodeJS types. Project C also builds fine by itself.

Project A references types from both B and C.

When attempting to build A, we get the error above. **But** the error is for project C.

Basically, when building Project A, the `skiplibcheck` option is being ignored and the incomplete NodeJS definition from the imported package is being used.

**Further Observations**
Curiously, if file `A/foo/bar.ts` is moved to `A/bar.ts` (and the imports amended) the build compiles fine.

**Attachment**
[Repro.zip](https://github.com/microsoft/TypeScript/files/4128129/Repro.zip)

**Comment by NoelAbrahams** [comment]
Just to say I've seen this https://github.com/microsoft/TypeScript/issues/23797#issuecomment-385818901

> `skipLibCheck` only implies that certain checks are performed on a declaration as necessary; when you need to compare two types, TypeScript decides to dive in and compare those types.

It's not clear that this applies to the case reported above. The behaviour is confusing. If a class in Project C builds fine by itself then referencing it in another project shouldn't report an error.

Furthermore, it's not clear why moving the file to a different folder should suppress the error.

**Comment by RyanCavanaugh** [comment]
I should note that `skipLibCheck` is more like a hint than an instruction. It means "don't exhaustively validate .d.ts files", **not** "do not report any errors in .d.ts files". It is entirely possible to write code that will cause the latter to happen even if `skibLibCheck` is on, though it doesn't sound like you've done that. If you *need* to not have an error reported in a .d.ts file in all cases, you need to use `ts-ignore` or an equivalent error suppression mechanism.
